src/auxiliary/mock_turbo.jl

```
@@ -0,0 +1,28 @@
+macro turbo(exprs...)
+    body = nothing
+    for expr in exprs
+        if Meta.isexpr(expr, :for)
+            body = expr
+        end
+    end
+    @assert body !== nothing
+
+    function insert_loopinfo!(expr)
+        recurse = Meta.isexpr(expr, :for) || Meta.isexpr(expr, :block) ||
+                  Meta.isexpr(expr, :let)
+        if recurse
+            foreach(insert_loopinfo!, expr.args)
+        end
+        if Meta.isexpr(expr, :for)
+            # TODO: Should we insert LLVM loopinfo or `julia.ivdep`?
+            push!(expr.args, Expr(:loopinfo, Symbol("julia.simdloop")))
+        end
```

Can you add a few comments here, please? Is this a public API, or are you doing something that may break at any time?

---

It's an internal API (https://github.com/JuliaLang/julia/blob/9f7cdfdc28290063ee021a5c6e180e82feee32c8/doc/src/devdocs/ast.md?plain=1#L464).

Most notably, it is used for the `@simd` macro https://github.com/JuliaLang/julia/blob/9f7cdfdc28290063ee021a5c6e180e82feee32c8/base/simdloop.jl#L79

There is also an old PR of mine that exposes some more of the knobs available through this API, that is used by a couple of consumers https://github.com/JuliaLang/julia/pull/31376

- https://github.com/JuliaSIMD/LLVMLoopInfo.jl/blob/main/src/LLVMLoopInfo.jl
- https://github.com/YingboMa/MaBLAS.jl/blob/master/src/loopinfo.jl
- https://github.com/JuliaGPU/KernelAbstractions.jl/blob/main/src/extras/loopinfo.jl

It is an internal API that directly interacts with the LLVM optimizer. It has not seen recent change, and I am not predicting any, worst case these annotations may be silently dropped.

---

Thanks. It's fine with me in this case. Could you please add a few comments/links to the code, check the TODO note, and let me know when this PR is finished from your point of view?